packages/appstore/src/components/modals/real-wallets-upgrade/real-wallets-upgrade.tsx

```diff
@@ -7,15 +7,15 @@ const RealWalletsUpgrade = observer(() => {
     const { traders_hub, ui } = useStore();
     const { is_real_wallets_upgrade_on, toggleWalletsUpgrade } = traders_hub;
     const { is_mobile } = ui;
-    const { startMigration, is_ineligible, is_migrating } = useWalletMigration();
+    const { startMigration, is_in_progress, is_ineligible, is_migrated, is_migrating } = useWalletMigration();
 
     const [current_step, setCurrentStep] = React.useState(0);
 
     React.useEffect(() => {
-        if (is_ineligible) {
+        if (is_in_progress || is_ineligible || is_migrated) {
             toggleWalletsUpgrade(false);
         }
-    }, [is_ineligible, toggleWalletsUpgrade]);
+    }, [is_in_progress, is_ineligible, is_migrated, toggleWalletsUpgrade]);
```

Comment:
The condition to exit the migration flow
Originally `!is_eligible`, neglected keeping it open when retrying from the `is_failed` state
Later `is_ineligible`, neglected closing it when a user has pressed the button to migrate

Now exiting the thing when non-eligible for it / already started it / already migrated